src/render/opengl/SDL_render_gl.c

```diff
@@ -843,6 +846,24 @@ static void GL_UnlockTexture(SDL_Renderer *renderer, SDL_Texture *texture)
     GL_UpdateTexture(renderer, texture, rect, pixels, data->pitch);
 }
 
+static int GL_GenMipmaps(SDL_Renderer *renderer, SDL_Texture *texture)
+{
+    GL_RenderData *renderdata = (GL_RenderData *)renderer->driverdata;
+    SDL_assert(renderdata->glGenerateMipmap != NULL);  // should have been set up at CreateRenderer time and we shouldn't be here if it isn't!
+
+    if (texture != renderdata->drawstate.texture) {
+        if (renderdata->GL_ARB_multitexture_supported) {
+            renderdata->glActiveTextureARB(GL_TEXTURE0_ARB);
+        }
+        const GL_TextureData *texturedata = (GL_TextureData *)texture->driverdata;
+        renderdata->glBindTexture(renderdata->textype, texturedata->texture);
+        renderdata->drawstate.texture = texture;
+    }
+
+    renderdata->glGenerateMipmap(renderdata->textype);
```

Comment:
Mipmaps need one of `GL_*_MIPMAP_*` modes (e.g `GL_LINEAR_MIPMAP_LINEAR`) for `GL_TEXTURE_MIN_FILTER` to have effect. See https://www.khronos.org/opengl/wiki/Common_Mistakes#Automatic_mipmap_generation